### ISSUE TITLE
Min-Warn operator

### ISSUE THREAD

**Issue by lippserd** [Icinga/icingaweb2-module-businessprocess#278]
Users may operate redundancy systems. If system A fails, system B takes over. In the event of an error, however, the overall status of both systems should be displayed as a warning, since a redundancy system is still running. If both systems are no longer there, it must be shown as critical. This is only possible to a very limited extent in the business process, since the best or worst available state is signaled. However, the status warning would be expected for ok and one or more warning/criticals.

I would have the following suggestion to switch business processes to warning status. We are implementing a new operator Min-Warn which is an extension of the Min operator.
Just a quick reminder of how the min operator works:

n of: a + b + c

The worst status from the best n states is used, e.g.

2 of: OK + OK + OK = OK
2 of: OK + OK + WARN = OK
2 of: OK + OK + CRIT = OK
2 of: OK + WARN + WARN = WARN
2 of: OK + WARN + CRIT = WARN
2 of OK + CRIT + CRIT = CRIT

Min operator (states, n): = worst state off (
  Best states from (states, n)
)

The extension looks like this:

n: m of: a + b + c

To put it simply, it specifies how many m states must be in Warning or Critical in order to switch the process to Warning status.

2: 1 of: OK + OK + OK = OK
2: 1 of: OK + OK + WARN = WARN
2: 1 of: OK + OK + CRIT = WARN
2: 1 of: OK + WARN + WARN = WARN
2: 1 of: OK + WARN + CRIT = WARN
2: 1 of OK + CRIT + CRIT = CRIT

Min warning operator (states, n, m): = worst state off (
  Min operator (states, n),
  Min operator (critical to warning (states), m)
)

**Comment by wp-perc** [comment]
Good! Several of my customers have a kind of scenario where this behavior can be applied.
+1 push

**Comment by giudig** [comment]
+1 Our customers needs this feature!

**Comment by gallowsky** [comment]
+1 we really need this feature
